websockets/websockets_test.go

```diff
@@ -879,3 +880,241 @@ func TestCookiesDefaultJar(t *testing.T) {
+			},
+			{
+				compression:   `null`,
+				expectedError: `unsupported compression algorithm 'null', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `undefined`,
+				expectedError: `unsupported compression algorithm 'undefined', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `" "`,
+				expectedError: `unsupported compression algorithm '', supported algorithm is 'deflate'`,
+			},
+			{compression: `"deflate"`},
+			{compression: `"deflate "`},
+			{
+				compression:   `"gzip"`,
+				expectedError: `unsupported compression algorithm 'gzip', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, gzip"`,
+				expectedError: `unsupported compression algorithm 'deflate, gzip', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, deflate"`,
+				expectedError: `unsupported compression algorithm 'deflate, deflate', supported algorithm is 'deflate'`,
+			},
+			{
+				compression:   `"deflate, "`,
+				expectedError: `unsupported compression algorithm 'deflate,', supported algorithm is 'deflate'`,
+			},
+		}
+
+		for _, testCase := range testCases {
+			testCase := testCase
+			t.Run(testCase.compression, func(t *testing.T) {
+				t.Parallel()
+				ts := newTestState(t)
+				sr := ts.tb.Replacer.Replace
+				ts.tb.Mux.HandleFunc("/ws-compression-param", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					upgrader := websocket.Upgrader{
+						EnableCompression: true,
+						ReadBufferSize:    1024,
+						WriteBufferSize:   1024,
+					}
+
+					conn, e := upgrader.Upgrade(w, req, w.Header())
+					if e != nil {
+						t.Fatalf("/ws-compression-param cannot upgrade request: %v", e)
+						return
+					}
+
+					e = conn.Close()
+					if e != nil {
+						t.Logf("error while closing connection in /ws-compression-param: %v", e)
+						return
+					}
+				}))
+
+				err := ts.ev.Start(func() error {
+					_, runErr := ts.rt.RunString(sr(`
+					var ws = new WebSocket("WSBIN_URL/ws-compression-param", null, {"compression":` + testCase.compression + `})
+					ws.onopen = () => {
+						ws.close()
+					}
+					`))
+					return runErr
+				})
+
+				if testCase.expectedError == "" {
+					require.NoError(t, err)
+				} else {
+					require.Error(t, err)
+					require.Contains(t, err.Error(), testCase.expectedError)
+				}
+			})
+		}
+	})
+}
+
+func clearSamples(tb *httpmultibin.HTTPMultiBin, samples chan metrics.SampleContainer) {
+	ctxDone := tb.Context.Done()
+	for {
+		select {
+		case <-samples:
+		case <-ctxDone:
+			return
+		}
+	}
+}
+
+func BenchmarkCompression(b *testing.B) {
+	const textMessage = 1
+	ts := newTestState(b)
+	sr := ts.tb.Replacer.Replace
+	go clearSamples(ts.tb, ts.samples)
+
+	testCodes := []string{
+		sr(`
+		var ws = new WebSocket("WSBIN_URL/ws-compression", null, {"compression":"deflate"})
+		ws.onmessage = () => {
+			ws.close()
```

Comment:
But the server sends messages, right?

```sh
Running tool: /home/olegbespalov/sdk/go1.19.3/bin/go test -benchmem -run=^$ -coverprofile=/tmp/vscode-goAaIBLy/go-code-cover -bench ^BenchmarkCompression$ github.com/grafana/xk6-websockets/websockets

goos: linux
goarch: amd64
pkg: github.com/grafana/xk6-websockets/websockets
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkCompression/compression-enabled-8         	    4440	    257307 ns/op	  116164 B/op	     623 allocs/op
BenchmarkCompression/compression-disabled-8        	    5079	    203712 ns/op	   68806 B/op	     582 allocs/op
PASS
coverage: 69.4% of statements
ok  	github.com/grafana/xk6-websockets/websockets	2.252s
```

Reply:
Well the server sends 1 message and then the connection is closed so :shrug: 

I am kind of for dropping this benchmark and maybe opening a PR to make some - although i would argue this will be mostly relevant once/if we decide to swap gorilla for something else :shrug:

Reply:
Sure, let's drop it for now :+1: